### PR TITLE
insights

### PR DESCRIPTION
**Pull request by gandesc**
have not yet split routes

### CODE REVIEW

File: src/app/Models/User.php [hunk 1]
```diff
@@ -169,13 +162,20 @@ public function delete()
 
         try {
             parent::delete();
-        } catch (\Exception $e) {
+        } catch (\Exception $exception) {
             throw new ConflictHttpException(__(
```
Comment (aocneanu) [inline src/app/Models/User.php]: add dedicated exception

File: src/app/Services/ProfileBuilder.php [hunk 2]
```diff
@@ -33,15 +33,15 @@ public function build()
...
         $this->user->rating = $this->rating();
     }
 
     private function rating()
     {
-        return intval((
-            self::LoginRating * $this->user->loginCount / $this->user->daysSinceMember +
-            self::ActionRating * $this->user->actionLogCount / $this->user->daysSinceMember
-        ) / 100);
+        $rating = self::LoginRating * $this->user->loginCount / $this->user->daysSinceMember +
```
Comment (aocneanu) [inline src/app/Services/ProfileBuilder.php]: implement `Decimals`

File: src/app/Models/User.php (hunk 1 above)
Comment (gandesc) [reply src/app/Models/User.php]: or remove the `delete` method and let the already imported `AvoidsDeletionConflicts` trait handle it?

Comment (aocneanu) [reply src/app/Models/User.php]: trait sounds good

**Pull request merged**